data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DynamicIndexManagerTests.java

```diff
@@ -0,0 +1,183 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class DynamicIndexManagerTests {
+
+    private static final String ID = "id";
+    private static final String DYNAMIC = "dynamic";
+    private static final String NEW_DYNAMIC = "new-dynamic";
+    private static final String INDEX_ALIAS = "test-${" + ID + "}-index-alias";
+    private static final String DATE_PATTERN = "yyyy.MM.dd";
+    private static final String INDEX_ALIAS_WITH_DATE_PATTERN = INDEX_ALIAS+ "-%{" + DATE_PATTERN + "}";
+
+    @Mock
+    private IndexManagerFactory mockIndexManagerFactory;
+    private IndexManagerFactory indexManagerFactory;
+
+    @Mock
+    private IndexManager indexManager;
+
+    private DynamicIndexManager dynamicIndexManager;
+
+    private IndexManager innerIndexManager;
+
+    @Mock
+    private RestHighLevelClient restHighLevelClient;
+
+    @Mock
+    private OpenSearchSinkConfiguration openSearchSinkConfiguration;
+
+    @Mock
+    private ClusterClient cluster;
+
+    @Mock
+    private ClusterGetSettingsResponse clusterGetSettingsResponse;
+
+    @Mock
+    private IndexConfiguration indexConfiguration;
+
+    @Mock
+    private IndicesClient indicesClient;
+
+    @Mock
+    private GetIndexTemplatesResponse getIndexTemplatesResponse;
+
+    @Mock
+    private RestClient restClient;
+
+    @Mock
+    private ResponseException responseException;
+
+    static final String EVENT_TYPE = "event";
+
+    @BeforeEach
+    public void setup() throws IOException {
+        initMocks(this);
+
+        indexManagerFactory = new IndexManagerFactory();
+        mockIndexManagerFactory = mock(IndexManagerFactory.class);
+        when(openSearchSinkConfiguration.getIndexConfiguration()).thenReturn(indexConfiguration);
+        when(indexConfiguration.getIsmPolicyFile()).thenReturn(Optional.empty());
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS);
+        when(restHighLevelClient.cluster()).thenReturn(cluster);
+        when(cluster.getSettings(any(ClusterGetSettingsRequest.class), any(RequestOptions.class)))
+                .thenReturn(clusterGetSettingsResponse);
+        when(restHighLevelClient.indices()).thenReturn(indicesClient);
+        dynamicIndexManager = new DynamicIndexManager(IndexType.CUSTOM, restHighLevelClient, openSearchSinkConfiguration, mockIndexManagerFactory);
+    }
+
+    @Test
+    public void dynamicIndexBasicTest(){
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS);
+        when(clusterGetSettingsResponse.getSetting(IndexConstants.ISM_ENABLED_SETTING)).thenReturn("true");
+        String configuredIndexAlias = openSearchSinkConfiguration.getIndexConfiguration().getIndexAlias();
+        try {
```

Comment:
Please remove these try-catch blocks (in lots of these tests).

You can add a throws clause to these test methods instead.

```
@Test
public void dynamicIndexBasicTest() throws IOException {
```